magicctapipe/scripts/lst1_magic/magic_calib_to_dl1.py

```diff
@@ -162,6 +167,38 @@ def magic_calib_to_dl1(input_file, output_dir, config, max_events, process_run=F
             subrun_id = event_source.metadata["subrun_number"][0]
             output_file = f"{output_dir}/dl1_M{tel_id}.Run{obs_id:08}.{subrun_id:03}.h5"
 
+    # Reset the telescope IDs of the subarray description
+    if event_source.is_hast:
+        tel_positions_magic_lst = {
+            1: [-8.09, 77.13, 0.78] * u.m,  # LST-1
+            2: [39.3, -62.55, -0.97] * u.m,  # MAGIC-I
+            3: [-31.21, -14.57, 0.2] * u.m,  # MAGIC-II
+        }
+
+        tel_descriptions_magic_lst = {
+            1: subarray.tel[1],  # dummy telescope description for LST-1
+            2: subarray.tel[1],  # MAGIC-I
+            3: subarray.tel[2],  # MAGIC-II
+        }
+
+        subarray_magic = SubarrayDescription(
+            "MAGIC-LST-Array", tel_positions_magic_lst, tel_descriptions_magic_lst
+        )
+    else:
```

Comment:
so if you have joint data but with a software trigger it will also only use MAGIC positions ?? something I do not understand here because the new code works with the 2019/20 data that are pre-HaST

Reply:
this was not changed wrt before. When we did not have HaST, only the MAGIC positions were written out in the subarray, with shift of index (1-->2, 2-->3, so that LST-1 was 1). Then the subarray information is changed again in the coincidence scripts (from line 562).

Reply:
ok, let's keep it like this for this PR, but I think we will need to revise it in the future, because the logic of the code is confusing